src/QuantumComputation.cpp

```diff
@@ -581,6 +581,339 @@ namespace qc {
+                }
+                counts[shot]++;
+            }
+
+            return counts;
+        }
+    }
+
+    void QuantumComputation::extractProbabilityVector(const VectorDD& in, std::vector<dd::fp>& probVector, std::unique_ptr<dd::Package>& dd) {
+        // ! initial layout, output permutation and garbage qubits are currently not supported here
+
+        dd->incRef(in);
+        std::map<std::size_t, char> measurements{};
+        // the number of classics dictates the size of the resulting probability vector
+        probVector = std::vector<dd::fp>(1ULL << nclassics);
+
+        extractProbabilityVectorRecursive(in, ops.begin(), measurements, 1., probVector, dd);
+    }
+
+    void QuantumComputation::extractProbabilityVectorRecursive(const VectorDD& currentState, decltype(ops.begin()) currentIt, std::map<std::size_t, char> measurements, dd::fp commonFactor, std::vector<dd::fp>& probVector, std::unique_ptr<dd::Package>& dd) {
+        auto state = currentState;
+        for (auto it = currentIt; it != ops.end(); ++it) {
+            auto& op = (*it);
+
+            // check whether a classic controlled operations can be applied
+            if (op->getType() == ClassicControlled) {
+                auto*       classicControlled = dynamic_cast<ClassicControlledOperation*>(op.get());
+                const auto& controlRegister   = classicControlled->getControlRegister();
+                const auto& expectedValue     = classicControlled->getExpectedValue();
+                auto        actualValue       = 0U;
+                // determine the actual value from measurements
+                for (std::size_t j = 0; j < controlRegister.second; ++j) {
+                    actualValue |= (measurements[controlRegister.first + j] == '1') ? (1U << j) : 0U;
+                }
+
+                // do not apply an operation if the value is not the expected one
+                if (actualValue != expectedValue)
+                    continue;
+            }
+
+            if (op->getType() == Reset) {
+                // a reset operation should only happen once a qubit has been measured, i.e., the qubit is in a basis state
+                // thus the probabilities for 0 and 1 need to be determined
+                // if p(1) = 1, an X operation has to be applied to the qubit
+                // if p(0) = 1, nothing has to be done
+                // if 0 < p(0), p(1) < 1, an error should be raised
+
+                const auto& targets = op->getTargets();
+                if (targets.size() != 1) {
+                    throw qc::QFRException("Resets on multiple qubits are currently not supported. Please split them into multiple single resets.");
+                }
+
+                const auto& [pzero, pone] = dd->determineMeasurementProbabilities(state, targets[0], true);
+
+                //                if (dd::ComplexTable<>::Entry::approximatelyOne(pone)) {
+                if (pone > 0.1) {
```

Comment:
Comment on why `0.1`?

Reply:
I fixed this to be more in line with what is described to happen in the code comment above.